File: app/src/main/java/com/innovation4you/napking/model/RestStop.java
package com.innovation4you.napking.model;

import java.util.Calendar;
import java.util.List;

public class RestStop {

	public long id;
	public String name;
	public double lat, lng;
	public int parkingSpaces;
	public String[] facilities;
	public List<OccupancyEntry> occupancies;

	public RestStop(final long id, final String name, final List<OccupancyEntry> occupancies, final double lat, final double lng) {
		this.id = id;
		this.name = name;
		this.occupancies = occupancies;
		this.lat = lat;
		this.lng = lng;
	}

	public int getOccupancyAtTime(final Calendar calendar) {
		final int weekday = calendar.get(Calendar.DAY_OF_WEEK) - 2;
		final int hour = calendar.get(Calendar.HOUR_OF_DAY);

		for (OccupancyEntry oe : occupancies) {
			if (oe.weekDay == weekday && oe.hour == hour) {
				return oe.occupancy;
			}
		}
		return 0;
	}

	@Override
	public String toString() {
		return name;
	}
}

File: app/src/main/java/com/innovation4you/napking/model/RestStop.kt
package com.innovation4you.napking.model

import java.util.*

class RestStop(var id: Long, var name: String, var occupancies: List<OccupancyEntry>, var lat: Double, var lng: Double) {
    var parkingSpaces: Int = 0
    var facilities: Array<String>? = null

    fun getOccupancyAtTime(calendar: Calendar): Int {
        val weekday = calendar.get(Calendar.DAY_OF_WEEK) - 2
        val hour = calendar.get(Calendar.HOUR_OF_DAY)

        for (oe in occupancies) {
            if (oe.weekDay == weekday && oe.hour == hour) {
                return oe.occupancy
            }
        }
        return 0
    }

    override fun toString(): String {
        return name
    }
}
